### PR TITLE
Retry git network operations

### LINKED ISSUE #418: git clone can fail
```
INFO:taskboot.target:Cloning https://github.com/MozillaSecurity/orion @ edb3e91c4deb1d7920fc7cd4171b16e6b9e43fe4
error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8)
error: 7671 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Traceback (most recent call last):
  File "/usr/bin/build", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/src/orion-builder/src/orion_builder/build.py", line 108, in main
    target = Target(args)
             ^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/taskboot/target.py", line 30, in __init__
  File "/usr/lib/python3.11/site-packages/taskboot/target.py", line 39, in clone
  File "/usr/lib/python3.11/subprocess.py", line 466, in check_output
  File "/usr/lib/python3.11/subprocess.py", line 571, in run
subprocess.CalledProcessError: Command '['git', 'clone', '--quiet', 'https://github.com/MozillaSecurity/orion', '/tmp/taskboot.o2wmkh17']' returned non-zero exit status 128.
```

I assume this is a network glitch, but regardless it should be retried like other network operations.

### PR DESCRIPTION
**Pull request by jschwartzentruber**
Fixes #418

I've tested `Target` locally using:

- `--git-repository https://github.com/MozillaSecurity/orion --git-revision master`
- `--git-repository https://github.com/MozillaSecurity/orion --git-revision edb3e91c4deb1d7920fc7cd4171b16e6b9e43fe4`

### CODE REVIEW

File: taskboot/target.py [hunk 1]
```diff
@@ -35,27 +37,16 @@ def clone(self, repository, revision):
...
-                stderr=subprocess.DEVNULL,
-            ).returncode
-            != 0
-        ):
-            cmd = ["git", "fetch", "--quiet", "origin", revision]
-            subprocess.check_output(cmd, cwd=self.dir)
-
         # Checkout revision to pull modifications
-        cmd = ["git", "checkout", revision, "-b", "taskboot"]
+        cmd = ["git", "checkout", "FETCH_HEAD", "-b", "taskboot"]
```
Comment (marco-c) [inline taskboot/target.py:49]: Nit: is this change necessary?

**Review by marco-c — APPROVED** [review]

File: taskboot/target.py (hunk 1 above)
Comment (jschwartzentruber) [reply taskboot/target.py:49]: It is. Without this, only revision hashes work, not names.

```
>>> from taskboot.target import Target
>>> from argparse import Namespace
>>> args = Namespace(git_repository="https://github.com/MozillaSecurity/orion", git_revision="master", target=None)
>>> Target(args)
fatal: 'master' is not a commit and a branch 'taskboot' cannot be created from it
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/truber/src/m/task-boot/taskboot/target.py", line 32, in __init__
    self.clone(args.git_repository, args.git_revision)
  File "/home/truber/src/m/task-boot/taskboot/target.py", line 50, in clone
    subprocess.check_output(cmd, cwd=self.dir)
  File "/usr/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'checkout', 'master', '-b', 'taskboot']' returned non-zero exit status 128.
```

Using `FETCH_HEAD`, both `master` or a git revision hash will work.

**Comment by marco-c** [comment]
Thanks!
